src/subdomains/generic/user/models/bank-data/bank-data.service.ts

```diff
@@ -17,6 +21,60 @@ export class BankDataService {
     private readonly specialAccountService: SpecialExternalAccountService,
   ) {}
 
+  @Cron(CronExpression.EVERY_MINUTE)
+  @Lock(1800)
+  async checkAndSetActive() {
+    if (DisabledProcess(Process.BANK_DATA_VERIFICATION)) return;
+    await this.bankDataVerification();
+  }
+
+  async bankDataVerification(): Promise<void> {
+    const entities = await this.bankDataRepo.find({
+      where: { active: IsNull(), type: Not(In([BankDataType.IDENT, BankDataType.USER])), comment: IsNull() },
+      relations: { userData: true },
+    });
+
+    for (const entity of entities) {
+      const existing = await this.bankDataRepo.findOne({
+        where: { iban: entity.iban, active: true },
+        relations: { userData: true },
+      });
+
+      const errors = this.getBankDataVerificationErrors(entity, existing);
+
+      if (errors.length === 0) {
+        if (!existing) {
+          await this.bankDataRepo.update(...entity.activate());
+        } else if (existing.type === BankDataType.BANK_IN || entity.type !== BankDataType.BANK_IN) {
+          await this.bankDataRepo.update(...entity.deactivate(BankDataVerificationError.ALREADY_ACTIVE_EXISTS));
+        } else {
+          const existingError = [...(existing.comment?.split(';') ?? []), BankDataVerificationError.NEW_BANK_IN_ACTIVE];
+
+          await this.bankDataRepo.update(...existing.deactivate(existingError.join(';')));
+          await this.bankDataRepo.update(...entity.activate());
+        }
+      } else {
+        if (existing.type === BankDataType.BANK_IN) errors.push(BankDataVerificationError.ALREADY_ACTIVE_EXISTS);
+
+        await this.bankDataRepo.update(...entity.deactivate(errors.join(';')));
+      }
```

Comment:
Can't we move the `existing.type === BankDataType.BANK_IN` check into the getErrors method?